packages/core/src/did/Did.utils.ts

```diff
@@ -21,25 +20,24 @@ import {
   SERVICE_KILT_MESSAGING,
 } from './Did'
 
-interface IEncodedDid extends Codec {
-  toJSON: () => [string, string, string | null] | null
+export interface IEncodedDidRecord extends Struct {
+  readonly signKey: Hash
+  readonly boxKey: Hash
+  readonly docRef: Option<Vec<u8>>
 }
 
 export function decodeDid(
   identifier: string,
-  encoded: Option<Tuple>
+  encoded: Option<IEncodedDidRecord>
 ): IDid | null {
-  DecoderUtils.assertCodecIsType(encoded, [
-    'Option<(PublicSigningKey,PublicBoxKey,Option<Bytes>)>',
-  ])
-  const decoded = (encoded as IEncodedDid).toJSON()
-  if (decoded) {
-    const documentStore = isHex(decoded[2]) ? hexToString(decoded[2]) : null
+  DecoderUtils.assertCodecIsType(encoded, ['Option<DidRecord>'])
+  if (encoded.isSome) {
+    const did = encoded.unwrap()
     return {
       identifier,
-      publicSigningKey: decoded[0],
-      publicBoxKey: decoded[1],
-      documentStore,
+      publicSigningKey: did.signKey.toString(),
+      publicBoxKey: did.boxKey.toString(),
+      documentStore: did.docRef.unwrapOrDefault().toHuman()?.toString(),
```

Comment:
doesn't this just build an string presentation of an array? why do we no longer use `hexToString`

Reply:
Good point, fixed in f494360

Reply:
I think to get to the target of `string | undefined` we even need to do something like 
```typescript
did.docRef.isSome ? hexToString(did.docRef.toHex()) : undefined
```
(sidetone: seems like `toHex()` handles a `none` option directly by returning `0x`)

Reply:
Fixed in a8496ed. I was wondering why we don't to any real integration tests with DIDs or was I not looking thoroughly enough? I wanted to test my and your proposal but did not want to invest time into setting up a test for this.